### PR TITLE
Support for "--headless-hosted-plugin-inspect" cmd argument 

### LINKED ISSUE #13462: Need support for a "--headless-hosted-plugin-inspect" backend cmdline argument
### Bug Description:

We need support for a `--headless-hosted-plugin-inspect` (or similar name) backend cmdline argument that will tell the headless plugin-host to listen on that port for a debug attachment. Today, both plugin hosts seem to be affected by the traditional `--hosted-plugin-inspect` argument, but clearly only one can listen on that port. So the first plugin-host to run gets the port and the other plugin-host becomes un-debuggable. As the headless plugin host always runs first, that's the winner.

Note: the notion of headless plugins was introduced in Theia 1.46.0.

**Additional Info**
Today I tried debugging a headless plugin using VS Code but in an environment focused on just the plugin developer instead of a Theia app developer that is also developing a plugin. Additionally, the plugin package being developed contributes both a standard plugin (runs in Node.js but interacts with app's frontend) and a headless plugin (runs in Node.js but interacts with app's backend).

Under the covers, there is a plugin-host process for the standard plugins (one for each connected frontend), and a single plugin-host process for all the headless plugins

The way you historically use VSCode to debug a plugin in this plugin-centric type of environment is to launch the app's backend using `--hosted-plugin-inspect=<port>`, then use a launch config in VS Code like this
```
{
    "type": "node",
    "request": "attach",
    "name": "Attach to Plugin Host",
    "port": <port>,
    ...
}
```

But as described earlier, only one of the two plugin-host processes will get to use the port specified via `--hosted-plugin-inspect`--the first plugin-host to run. 

We should have a switch specifically for the headless plugin-host, reserving the current one for the standard plugin host.

### Steps to Reproduce:
```
git clone -b std_and_headless_plugins https://github.com/jcortell68/theiasandbox
cd theiasandbox
yarn
yarn start:electron --hosted-plugin-inspect=9339
```

1. In VS Code, launch the `Attach to Plugin Host` launch configuration.
2. Set a breakpoint on the "heartbeat" lines of 
- myext\src\extension-headless.ts
- myext\src\extension.ts
3.  Notice that only the `extension-headless.ts` breakpoint resolves and is hit

To prove that both plugin-hosts competed for the debug port,  remove these lines from `package.json`
````
  "theiaPlugin": {
    "headless": "./dist/extension-headless.js"
  },
````
That will make the plugin-package have just a standard plugin. You'll see that its breakpoint is now hit.

As further proof, the log/console has this:
```
2024-03-14T13:10:04.417Z root ERROR [hosted-plugin: 15948] Starting inspector on 127.0.0.1:9339 failed: address already in use
```
### Additional Information

- Operating System: Windows
- Theia Version: 1.47.1

### PR DESCRIPTION
**Pull request by jfaltermeier**
#### What it does

* add names for plugin servers that will be used on the command line like `--server-name-inspect`
* enable `--headless-hosted-plugin-inspect`
* add plugin-dev preferences for specifying ports per plugin-server via the new server-name

closes #13462 

#### How to test

See instructions on the ticket https://github.com/eclipse-theia/theia/issues/13462

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Change looks good to me and works very well, thank you Johannes! I do have a few questions and some small nitpicks but nothing major.

File: packages/plugin-ext/src/hosted/node/plugin-service.ts [hunk 1]
```diff
@@ -75,6 +77,8 @@ export class HostedPluginServerImpl implements HostedPluginServer {
 
     @postConstruct()
     protected init(): void {
+        this.serverName = this.getServerName();
```
[the remark this answers is not in the document]
Comment (jfaltermeier) [reply packages/plugin-ext/src/hosted/node/plugin-service.ts]: I'll remove the property

File: packages/plugin-dev/src/browser/hosted-plugin-manager-client.ts [hunk 2]
```diff
@@ -248,7 +248,8 @@ export class HostedPluginManagerClient {
                 try {
                     if (this.isDebug) {
                         this.pluginInstanceURL = await this.hostedPluginServer.runDebugHostedPluginInstance(this.pluginLocation!.toString(), {
-                            debugMode: this.hostedPluginPreferences['hosted-plugin.debugMode']
+                            debugMode: this.hostedPluginPreferences['hosted-plugin.debugMode'],
+                            debugPort: [... this.hostedPluginPreferences['hosted-plugin.debugPorts']]
```
[the remark this answers is not in the document]
Comment (jfaltermeier) [reply packages/plugin-dev/src/browser/hosted-plugin-manager-client.ts]: It should be the "Hosted Plugin: Restart Instance" command, but I think this does not corretely identify whether there is a running hosted plugin. It always says "Hosted Plugin: Starting" for me (also before my changes) .

[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply packages/plugin-dev/src/browser/hosted-plugin-manager-client.ts]: Right, if that is the case I think we can ignore it for this PR. Could you please log an issue for that?

**Review by martin-fleck-at — APPROVED** [review]
Thank you for the quick update! Change looks good to me.

**Pull request merged**